cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -445,6 +447,22 @@ public void testGenerateHeadlessServiceWithJmxMetrics() {
         checkOwnerReference(kc.createOwnerReference(), headless);
     }
 
+    @ParallelTest
+    public void testExposesJmxContainerPortWhenJmxEnabled() {
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editSpec()
+                    .editKafka()
+                        .withJmxOptions(new KafkaJmxOptionsBuilder().build())
+                    .endKafka()
+                .endSpec()
+                .build();
+        KafkaCluster kc = KafkaCluster.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafka, VERSIONS);
+        ContainerPort jmxContainerPort = kc.createContainerPort(JMX_PORT_NAME, JMX_PORT, "TCP");
+        List<Container> containers = kc.getContainers(ImagePullPolicy.IFNOTPRESENT);
```

Comment:
I am not sure this test is right. You are creating a `KafkaCluster` model from the `Kafka` resource where you enabled JMX which would be enough to set `isJmxEnabled` to true internally. Then you create the container port on the resource, so actually "forcing" the port creation while it should happen because of the set in the resource.
Instead you should just get the container port list via `getContainerPortList()` right after the `fromCrd` and verify that the JMX port is present in that list.

Reply:
I'm creating `jmxContainerPort` so that and I can check that the list of container ports contains that port. I had tested this by commenting out the new conditional in `KafkaCluster` which adds the JMX port. When the conditional is commented out then this test fails. Would it be clearer if we moved `jmxContianerPort` down where the assertion happens?

Reply:
Using the `getContainerPortList()` also checks that the `isJmxEnabled` flag is set correctly by the `fromCrd` (depending on the `Kafka` CR) and that based on that flag, the JMX port is created.

Reply:
Ah sorry, I got what you mean now! But moving it later sounds better yes.

Reply:
Are you suggesting to call `kc.getContainerPortList()` instead of `kc.getContainers()`? I went to make this change but `getContainerPortList` is private on `ZookeeperCluster`.

I think we also commented at the same time so wasn't sure if you saw my previous message.

Reply:
Yes we commented at same time. Calling `getContainers` is right because it's going to call `getContainerPortList`. I would just create the JMX port right before the assert for clarity. Anyway I already approved it ;-)

Reply:
I went ahead and moved `jmxContainerPort` next to the assertion. Hopefully that improves the clarity of what this test is asserting.

Reply:
Thanks @michael-barker !

Reply:
:rofl: Did it again. At least we're on the same page when commenting at the same time.